core/src/main/scala/org/apache/spark/rpc/netty/Outbox.scala

```diff
@@ -215,7 +216,7 @@ private[netty] class Outbox(nettyEnv: NettyRpcEnv, val address: RpcAddress) {
     // update messages and it's safe to just drain the queue.
     var message = messages.poll()
```

Comment:
I doubt there will be many messages in the outbox at a time. This code mostly fixes a race during initialization of the RPC channel, later the outbox would be mostly a bypass. #9210 even bypasses the outbox altogether in a lot of cases.